number.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package core
+package otel
```

Comment:
Should this go to its own package the same way as `KeyValue`? 

`number.NewInt64()` is better than `otel.NewInt64Number()`